### PR TITLE
java: add headers to http transport

### PR DESCRIPTION
**Pull request by tnazarew**
### Problem

There is no way to supply custom headers while emitting events via http. 

Closes: #1712 

### Solution

Add new field to HttpConfig to pass custom headers to client. Add handling for another property-like field in spark integration.

> **Note:** All schema changes require discussion. Please [link the issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for context.

- [ ] Your change modifies the [core](https://github.com/OpenLineage/OpenLineage/blob/main/spec/OpenLineage.json) OpenLineage model
- [ ] Your change modifies one or more OpenLineage [facets](https://github.com/OpenLineage/OpenLineage/tree/main/spec/facets)

If you're contributing a new integration, please specify the scope of the integration and how/where it has been tested (e.g., Apache Spark integration supports `S3` and `GCS` filesystem operations, tested with AWS EMR).

### Checklist

- [x] You've [signed-off](https://github.com/OpenLineage/OpenLineage/blob/main/why-the-dco.md) your work
- [x] Your pull request title follows our [guidelines](https://github.com/OpenLineage/OpenLineage/blob/main/CONTRIBUTING.md#creating-pull-requests)
- [x] Your changes are accompanied by tests (_if relevant_)
- [x] Your change contains a [small diff](https://kurtisnusbaum.medium.com/stacked-diffs-keeping-phabricator-diffs-small-d9964f4dcfa6) and is self-contained
- [x] You've updated any relevant documentation (_if relevant_)
- [x] Your comment includes a one-liner for the changelog about the specific purpose of the change (_if necessary_)
- [ ] You've versioned the core OpenLineage model or facets according to [SchemaVer](https://docs.snowplowanalytics.com/docs/pipeline-components-and-applications/iglu/common-architecture/schemaver) (_if relevant_)
- [x] You've added a [header](https://github.com/OpenLineage/OpenLineage/tree/main/.github/header_templates.md) to source files (_if relevant_)

----
SPDX-License-Identifier: Apache-2.0\
Copyright 2018-2023 contributors to the OpenLineage project

### CODE REVIEW

File: client/java/src/test/java/io/openlineage/client/transports/HttpTransportTest.java [hunk 1]
```diff
@@ -209,4 +218,42 @@ void clientClosesNetworkResources() throws IOException {
...
+    config.setHeaders(headers);
+
+    CloseableHttpClient http = mock(CloseableHttpClient.class);
+    Transport transport = new HttpTransport(http, config);
+    OpenLineageClient client = new OpenLineageClient(transport);
+
+    CloseableHttpResponse response = mock(CloseableHttpResponse.class, RETURNS_DEEP_STUBS);
+    when(response.getStatusLine().getStatusCode()).thenReturn(200);
+    when(response.getEntity().isStreaming()).thenReturn(true);
+    Map<String, HttpPost> map = new HashMap<>();
```
Comment (pawel-big-lebowski) [inline client/java/src/test/java/io/openlineage/client/transports/HttpTransportTest.java:241]: Isn't it just a map with a singe entry with a key `test`? If so, why do we need it?

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java [hunk 2]
```diff
@@ -99,6 +101,8 @@ void testConfToHttpConfig() {
     assert (transportConfig.getAuth() instanceof ApiKeyTokenProvider);
     assertEquals("Bearer random_token", transportConfig.getAuth().getToken());
     assertEquals(5000, transportConfig.getTimeout());
+    assertEquals("test1", transportConfig.getHeaders().get("test1"));
```
Comment (pawel-big-lebowski) [inline integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java:104]: I think it's a good practice to set different values for keys and values.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,8 @@
 ### Added
 * **Spark: spark properties facet** [`#1717`](https://github.com/OpenLineage/OpenLineage/pull/1717) by [@tnazarew](https://github.com/tnazarew)    
   *Add facet to capture specified Spark properties*
+* **Java: add headers to http transport** [`#1718`](https://github.com/OpenLineage/OpenLineage/pull/1718) by [@tnazarew](https://github.com/tnazarew)
```
Comment (pawel-big-lebowski) [inline CHANGELOG.md:8]: Consider `add configurable headers`

**Review by pawel-big-lebowski — COMMENTED** [review]
Looks good to me 🚀 
Some minor comments added.

File: integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java (hunk 2 above)
Comment (tnazarew) [reply integration/spark/app/src/test/java/io/openlineage/spark/agent/ArgumentParserTest.java:104]: ok I'll change that

File: CHANGELOG.md (hunk 3 above)
Comment (tnazarew) [reply CHANGELOG.md:8]: yup, makes sense

**Review by pawel-big-lebowski — APPROVED** [review]

**Pull request merged**